app/services/foreman/plugin.rb

```diff
@@ -254,8 +255,8 @@ def allowed_template_variables(*variables)
 
     # Add Compute resource
     def compute_resource(provider)
-      SETTINGS[provider.name.split('::').last.downcase.to_sym] = true
-      ComputeResource.register_provider provider
+      return if @compute_resources.include?(provider)
+      @compute_resources << provider
```

Comment:
Storing a `provider.name` instead of the `provider` would prevent potential issue of somebody using an out-dated reference when the class gets reloaded. Not a big deal though.